Trident v21.04 templates added (#118)
IBM/ibm-satellite-storage: config-templates/netapp/netapp-ontap-nas/21.04/README.md

# NetApp Trident - ONTAP NAS Driver

> You can use the `netapp-ontap-nas` Satellite storage template to deploy NetApp storage drives which you can use dynamically provision and mange file volumes in your ONTAP NAS.

Reviewer: ```suggestion
You can use the `netapp-ontap-nas` Satellite storage template to deploy NetApp storage drivers which you can use dynamically provision and manage file volumes in your ONTAP NAS.
```

## Prerequisites

**Planning consideration for Infra Admin**
* Create a cluster that meets the requirements for ONTAP NAS. For more information, see the [NetApp documentation](https://netapp-trident.readthedocs.io/en/stable-v21.04/support/requirements.html). Verify that your backend ONTAP cluster is configured as a Trident backend.
   * You must have a dedicated SVM for Trident. Volumes that are created by Trident are created in this SVM.
   * You must have one or more aggregates assigned to the SVM.
   * You must have one or more dataLIFs for the SVM. Depending on the protocol used (NFS/iSCSI), at least one dataLIF is required.
   * You must have NFS services enabled on the SVM.
   * You must set up a snapshot policy on the SVM.
* Share following details with Location Admin

> **Planning consideration for Location Admin**

Reviewer: ```suggestion
**Planning considerations for the Satellite Location Admin**
```

| `nfsMountOptions` | Optional | Specify the NFS mount version. Example: `nfsvers=4` | `nfsvers=4` |

## Default storage classes

The following storage classes are installed when you assign your `netapp-ontap-nas` configuration to your clusters to allow you to take advantage of ONTAP's QoS features:

| Storage class name | Type | File system | IOPs | Reclaim policy |
| --- | --- | --- | --- | --- |
| `ntap-file-gold` | Ontap-NAS | NFS | user defined**\*** | Delete |
| `ntap-file-silver` | Ontap-NAS | NFS | user defined**\*** | Delete |
| `ntap-file-bronze` | Ontap-NAS | NFS | user defined**\*** | Delete | 
| `ntap-file-default` | Ontap-NAS | NFS | n/a | Delete | 

**\*NOTE**: In order to use the **ntap-file-gold**, **ntap-file-silver** or **ntap-file-bronze** storage classes, the storage administrator must create the associated QoS policies on the storage controller using the following QoS Policy Group names: **gold**, **silver**, **bronze**.

> For information on creating and managing QoS Policy groups, please refer to the [ONTAP 9 Storage Management documentation](https://docs.netapp.com/ontap-9/index.jsp).

Reviewer: ```suggestion
For information on creating and managing QoS Policy groups, please refer to the [ONTAP 9 Storage Management documentation](https://docs.netapp.com/ontap-9/index.jsp).

```

| `password` | Required | The password to connect to the storage device. | N/A |
| `limitVolumeSize` | Optional | Maximum volume size that can be requested and qtree parent volume size. Example: `50Gi` | `(not enforced by default)` |
| `limitAggregateUsage` | Optional | Limit provisioning of volumes if parent volume usage exceeds this value. For example, if a volume is requested that causes parent volume usage to exceed this value, the volume provisioning fails. Example: `80%` | `(not enforced by default)` |
| `nfsMountOptions` | Optional | Specify the NFS mount version. Example: `nfsvers=4` | `""` |

## Default storage classes

The following storage classes are installed when you assign your `netapp-ontap-nas` configuration to your clusters to allow you to take advantage of ONTAP's QoS features:

| Storage class name | Type | File system | IOPs | Reclaim policy |
| --- | --- | --- | --- | --- |
| `ntap-file-gold` | Ontap-NAS | NFS | user defined**\*** | Delete |
| `ntap-file-silver` | Ontap-NAS | NFS | user defined**\*** | Delete |
| `ntap-file-bronze` | Ontap-NAS | NFS | user defined**\*** | Delete | 
| `ntap-file-default` | Ontap-NAS | NFS | n/a | Delete | 

> 

Reviewer 2: Update the storage class names to `sat-netapp-file-gold`,`sat-netapp-file-silver`, `sat-netapp-file-bronze` and `sat-netapp-file-default`
